api/v1.inc

```diff
@@ -31,6 +35,12 @@ $router->add_route("GET", "v1/projects/imagesources", "api_v1_projects_imagesour
 $router->add_route("GET", "v1/projects/:projectid/pages", "api_v1_project_pages");
 $router->add_route("GET", "v1/projects/:projectid/pagedetails", "api_v1_project_pagedetails");
 $router->add_route("GET", "v1/projects/:projectid/pages/:pagename/pagerounds/:pageroundid", "api_v1_project_page_round");
+$router->add_route("POST", "v1/projects/:projectid/projectstates/:projectstateid/checkout", "api_v1_project_state_checkout");
```

Comment:
I don't think we want to couple any of the update APIs to the page state. The state is an intrinsic part of the page and not a subselector. The current interface passes the state around to ensure that someone doesn't click on a link to update a page that has changed state from what was expected, but we shouldn't try to codify that within the API definitions.

I'm envisioning that the APIs that we will need are:
```
# save page (mirrors the GET)
POST v1/projects/:projectid/pages/:pagename/pagerounds/:pageroundid

# get the next page to proof; marks a page for the user
# and returns the info needed to load the page (page name, round)
POST v1/projects/:projectid/pages/proofread (or /checkout or /next or something similar)
```

To save a page or return a page, we use the same "save page" function and pass in the state (ie save, done, return).

Reply:
I see what you mean about the page state.
The checkout you envisage is very close to what I've done.

Reply:
> The checkout you envisage is very close to what I've done.

Yes, exactly.